src/app/Shared/Services/NotificationChannel.service.tsx

```diff
@@ -82,40 +145,29 @@ export class NotificationChannel {
     private readonly notifications: Notifications,
     private readonly login: LoginService
   ) {
-    this.messages(NotificationCategory.WsClientActivity).subscribe(v => {
-      const addr = Object.keys(v.message)[0];
-      const status = v.message[addr];
-      notifications.info('WebSocket Client Activity', `Client at ${addr} ${status}`, NotificationCategory.WsClientActivity);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingCreated).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Created', `${event.recording} created in target: ${event.target}`);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingStopped).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Stopped', `${event.recording} was stopped`);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingSaved).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Archived', `${event.recording.name} was archived`);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingDeleted).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Deleted', `${event.recording} was deleted`);
-    });
-
-    this.messages(NotificationCategory.ArchivedRecordingCreated).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Archived', `${event.recording.name} was uploaded into archives`);
+    messageKeys.forEach((value, key) => {
+      if (!value) {
+        return;
+      }
+      this.messages(key).subscribe((msg: NotificationMessage) => {
+        const message = value.body.call(this, msg);
+        notifications.notify({ title: value.title, message, category: key, variant: value.variant })
+      });
     });
+    this._messages.pipe(
+      filter(msg => !messageKeys.has(msg.meta.category as NotificationCategory))
+    ).subscribe(msg => {
+      const category = NotificationCategory[msg.meta.category as keyof typeof NotificationCategory];
```

Comment:
Since this subscription handles messages which don't belong to any predefined field in the enum `NotificationCategory`, is this line simply saying "take whatever category `msg.meta.category` is and represent it as a new field in the enum `NotificationCategory`"?

Reply:
> Since this subscription handles messages which don't belong to any predefined field in the enum NotificationCategory

Not quite - it handles messages for which there is no defined Mapper configuration object. There *should* be a 1:1 relation between NotificationCategory and Mappers, but this is meant to handle any cases where there is a category missing a mapper.

> is this line simply saying "take whatever category msg.meta.category is and represent it as a new field in the enum NotificationCategory"?

It's a typecast of the raw string to a NotificationCategory. If there is no matching enum member of the NotificationCategory equal to the raw string then this still ends up being the same raw string (`msg.meta.category`). Really nothing happens to the raw string value in the end, but this just lets me pretend it has been cast to an enum value so that I can compare it to the enum values (which are really just syntax sugar over string constants here) in the `if-else` for setting the alert variant.

Reply:
Ah, okay that clears things up.